web/app/view/MapController.js

```diff
@@ -276,21 +278,62 @@ Ext.define('Traccar.view.MapController', {
     },
 
     selectDevice: function (device, center) {
+        this.hidePopup();
         this.selectMarker(this.latestMarkers[device.get('id')], center);
     },
 
     selectReport: function (position, center) {
         this.selectMarker(this.reportMarkers[position.get('id')], center);
     },
 
+    clickDevice: function (feature) {
+        var record, attributes;
+
+        record = feature.get('record');
+        attributes = record.get('attributes');
+        if (Object.getOwnPropertyNames(attributes).length) {
+            this.showAttributesView(feature);
+        }
+    },
+
     selectFeature: function (feature) {
         var record = feature.get('record');
         if (record) {
             if (record instanceof Traccar.model.Device) {
                 this.fireEvent('selectDevice', record, false);
+                this.fireEvent('clickDevice', feature);
             } else {
                 this.fireEvent('selectReport', record, false);
             }
         }
+    },
+
+    clickMap: function () {
+        this.hidePopup();
+    },
+
+    showAttributesView: function (feature) {
+        var coordinates, popupOverlay, record;
+
+        coordinates = feature.getGeometry().getCoordinates();
+        popupOverlay = this.getView().getPopupOverlay();
+        record = feature.get('record');
+
+        popupOverlay.setPosition(coordinates);
+
+        Ext.create('Traccar.view.PopupWindow', {
+            title: record.get('name') + ' ' + Strings.sharedAttributes,
+            renderTo: Ext.get('popup'),
+            items: {
+                xtype: 'attributesView',
+                record: record,
+                tbar: null
+            }
+        }).show();
+    },
+
+    hidePopup: function () {
+        var popupOverlay = this.getView().getPopupOverlay();
+        popupOverlay.setPosition(undefined);
```

Comment:
Is this the correct way of hiding a popup? Also, even if it is, you don't need to specify "undefined" attribute.